contributors/design-proposals/hugepages.md

```diff
@@ -0,0 +1,281 @@
+  Issues).
+
+1. A pod can use huge pages with any of the prior described methods.
+1. A pod can request huge pages.
+1. A scheduler can bind pods to nodes that have available huge pages.
+1. A quota may limit usage of huge pages.
+1. A limit range may constrain min and max huge page requests.
+
+## Feature Gate
+
+The proposal introduces huge pages as an Alpha feature.
+
+It must be enabled via the `--feature-gates=HugePages=true` flag on pertient
+components pending graduation to Beta.
+
+## Node Specfication
+
+Huge pages cannot be overcommitted on a node.
+
+A system may support multiple huge page sizes.  It is assumed that most
+nodes will be configured to primarily use the default huge page size as
+returned via `grep Hugepagesize /proc/meminfo`.  This defaults to 2MB on
+most Linux systems unless overriden by `default_hugepagesz=1g` in kernel
+boot parameters.  This design does not limit the ability to use multiple
+huge page sizes at the same time, but we expect this to not be the normal
+setup.  For each supported huge page size, the node will advertise a
+resource of the form `hugepages.<hugepagesize>`.  For example, if a node
+supports 2MB huge pages, a resource `hugepages.2MB` will be shown in node
+capacity and allocatable values.
+
+There are a variety of huge page sizes supported across different hardware
+architectures.  It is preferred to have a resource per size in order to
+better support quota.  For example, 1 huge page with size 2MB is an order
+of magnitude different than 1 huge page with size 1GB.  We assume gigantic
+pages are even more precious resources than huge pages.
+
+Pre-allocated huge pages reduce the amount of allocatable memory on a node. The
+node will treat pre-allocated huge pages similar to other system reservations
+and reduce the amount of `memory` it reports using the following formula:
+
+```
+[Allocatale] = [Node Capacity] - 
+ [Kube-Reserved] - 
+ [System-Reserved] - 
+ [Pre-Allocated-HugePages * HugePageSize] -
+ [Hard-Eviction-Threshold]
+```
+
+The following represents a machine with 10Gi of memory.  1Gi of memory has been
+reserved as 512 pre-allocated huge pages sized 2MB.  As you can see, the
+allocatable memory has been reduced to account for the amount of huge pages
+reserved.
+
+```
+apiVersion: v1
+kind: Node
+metadata:
+  name: node1
+...
+status:
+  capacity:
+    memory: 10Gi
+    hugepages.2MB: 512
+  allocatable:
+    memory: 9Gi
+    hugepages.2MB: 512
+...  
+```
+
+## Pod Specification
+
+A pod must make a request to consume pre-allocated huge pages using the resource
+`hugepages.<hugepagesize>` whose quantity is a non-negative number of pages. The
+request and limit for `hugepages.<hugepagesize>` must match until such time the
+node provides more reliability in handling overcommit (potentially via
+eviction).  An application that consumes `hugepages.<hugepagesize>` is not in
+the `BestEffort` QoS class.
```

Comment:
not sure if we need this sentence.  If we do need it, i'd go with

An application that consumes `hugepages.<hugepagesize>` is, by definition, not in the `BestEffort` QoS class as it specifies a request for one or more resources.